python/sglang/srt/managers/scheduler.py

```diff
@@ -683,11 +694,22 @@ def handle_embedding_request(
 
         # Truncate prompts that are too long
         if len(req.origin_input_ids) >= self.max_req_input_len:
-            logger.warning(
-                "Request length is longer than the KV cache pool size or "
-                "the max context length. Truncated!!!"
-            )
-            req.origin_input_ids = req.origin_input_ids[: self.max_req_input_len]
+            if self.server_args.allow_auto_truncate:
```

Comment:
wrap this into a utility function and reuse the code.

Reply:
Added it in utils.py